### PR TITLE
[JFG-988] Change validators distribution from reflection to providers

### PR DESCRIPTION
**Pull request by a-badaev**

### CODE REVIEW

File: chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java [hunk 1]
```diff
@@ -112,30 +113,30 @@ public Builder withInvoker(Class<? extends Invoker> invoker) {
...
          * Instances of those subtypes will be used to validate responses during Jagger test execution @n
          * Example:
          * @code
-         *      addValidators(Arrays.asList(com.griddynamics.jagger.engine.e1.collector.NotNullResponseValidator.class))
+         *      addValidator(ResponseValidatorProvider.of(NotNullResponseValidator.class))
          * @endcode
          * @see com.griddynamics.jagger.engine.e1.collector.NotNullResponseValidator for example
          */
-        public Builder addValidators(List<Class<? extends ResponseValidator>> validators) {
+        public Builder addValidators(List<KernelSideObjectProvider<ResponseValidator<Object, Object, Object>>> validators) {
```
Comment (oskliarov) [inline chassis/core/src/main/java/com/griddynamics/jagger/user/test/configurations/JTestDefinition.java]: What about to remove <Object, Object, Object> and make signature of this method like addListeners.
Also I propose create interface like
`interface ValidatorProvider<Q, E, R> implements KernelSideObjectProvider<ResponseValidator<Q, E, R>>`
to make code more readable and user friendly %) 
And add some example with custom provider, for example based on TrueValidator

File: chassis/core/src/main/java/com/griddynamics/jagger/engine/e1/collector/ResponseValidatorProvider.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.griddynamics.jagger.engine.e1.scenario.KernelSideObjectProvider;
+
+/**
+ * Provides instances of a ${@link ResponseValidator} class
+ * @n
+ * Created by Andrey Badaev
+ * Date: 14/12/16
+ */
+public interface ResponseValidatorProvider extends KernelSideObjectProvider<ResponseValidator<Object,Object,Object>> {
```
Comment (oskliarov) [inline chassis/core/src/main/java/com/griddynamics/jagger/engine/e1/collector/ResponseValidatorProvider.java]: will the following validator work with this interface
```
...
public boolean validate(JHttpQuery<String> query, JHttpEndpoint endpoint, JHttpResponse<MetricEntity[]> result, long duration) {
        List<MetricEntity> actualEntities = Arrays.asList(result.getBody());
....
```

why not `interface ResponseValidatorProvider<Q, E, R> extends  KernelSideObjectProvider<ResponseValidator<Q, E, R>>`

File: test/jagger-test/src/main/java/metric/TrueValidator.java [hunk 3]
```diff
@@ -2,15 +2,16 @@
...
 
 /**
- * Created with IntelliJ IDEA.
  * User: kgribov
  * Date: 8/16/13
  * Time: 6:26 PM
- * To change this template use File | Settings | File Templates.
  */
-public class TrueValidator extends ResponseValidator {
+public class TrueValidator extends ResponseValidator<JHttpQuery, JHttpEndpoint, JHttpResponse> {
```
Comment (dlatnikov) [inline test/jagger-test/src/main/java/metric/TrueValidator.java]: let's leave this one without changes

Comment (a-badaev) [inline test/jagger-test/src/main/java/metric/TrueValidator.java]: done

**Pull request merged**